File: pinyougou-cart-interface/src/main/java/com/lsf/pinyougou/cart/service/CartService.java
package com.lsf.pinyougou.cart.service;


import com.lsf.pinyougou.pojogroup.Cart;
import java.util.List;


public interface CartService {


    /**
     * 添加商品 SKU 到购物车
     *
     * @param cartList  旧的购物车列表
     * @param itemId    商品 SKU ID
     * @param num       购买数量
     *
     * @return          新的购物车列表
     */
    List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

}

File: pinyougou-seckill-web/src/main/webapp/js/service/cartService.js
// 购物车服务层
pyg.service('cartService', function ($http) {

    // 查询 cookie 中存储的购物车列表
    this.findCartList = function () {
        return $http.get('cart/findCartList.do');
    };


    // 添加商品 SKU 到购物车
    this.addGoodsToCartList = function (itemId, num) {
        return $http.get('cart/addGoodsToCartList.do?itemId=' + itemId + '&num=' + num);
    };


    // 计算商品总数量和总价
    this.countTotalValue = function (cartList) {
        var totalValue = { 'totalNum': 0, 'totalPrice': 0.00 };

        for (var i = 0; i < cartList.length; i++) {
            var cart = cartList[i];
            for (var j = 0; j < cart.orderItemList.length; j++) {
                var orderItem = cart.orderItemList[j];
                totalValue.totalNum += orderItem.num;
                totalValue.totalPrice += orderItem.totalFee;
            }
        }

        return totalValue;
    };

});
